adafruit_bh1750.py

```diff
@@ -97,15 +104,15 @@ class RWBitfields:
 
     """
 
-    def __init__(self, num_bits, lowest_bit):
+    def __init__(self, num_bits: int, lowest_bit: int) -> None:
         self._bit_mask = ((1 << num_bits) - 1) << lowest_bit
         self._lowest_bit = lowest_bit
 
-    def __get__(self, obj, objtype=None):
+    def __get__(self, obj: object, objtype: Optional[type] = None) -> int:
```

Comment:
A few things:

- A getter can be used on both instances and classes, which is what changes what `obj` will be.  It can either be the instance that "triggered" this getter, or `None` if used on a class.
- Since `RWBitfields` will never be used outside of `BH1750`, we can type annotate `obj` as a forward reference of `"BH1750"` when it's an instance.
- `objtype` will always be a type, regardless of whether instanced via instance or class, so it will never be `None`.
- Since this will always be the *type* of `BH1750` (not an instance of it), we can annotate it as such using `Type["BH1750"]`, with `Type` impored from `typing`.

All together this should be:

```suggestion
    def __get__(self, obj: Optional["BH1750"], objtype: Type["BH1750"]) -> int:
```

Here's an example to show what I'm talking about:

```python
class GetSetClass:

    def __init__(self, number):
        self.number = number

    def __get__(self, obj, objtype):
        print("GET:")
        print(f"Type of obj: {type(obj)}")
        print(f"Type of objtype: {type(objtype)}")
        return str(self.number)

    def __set__(self, obj, value):
        print("SET:")
        print(f"Type of obj: {type(obj)}")
        self.number = value

class DriverClass:

    attribute = GetSetClass(5)

driver = DriverClass()
print("Testing instance get:::")
attribute_as_instance = driver.attribute
print("Testing instance set:::")
driver.attribute = "7"
print("Testing class get:::")
attrbute_as_class = DriverClass.attribute

# This all prints the follwing:
#
# Testing instance get:::
# GET:
# Type of obj: <class '__main__.DriverClass'>
# Type of objtype: <class 'type'>
# Testing instance set:::
# SET:
# Type of obj: <class '__main__.DriverClass'>
# Testing class get:::
# GET:
# Type of obj: <class 'NoneType'>
# Type of objtype: <class 'type'>
```

Reply:
thanks for the education!